packages/provider/src/fetchai/proxy.ts

```diff
@@ -0,0 +1,81 @@
+import { JSONUint8Array, Result } from "@keplr-wallet/router";
+import { Method } from "./types";
+
+export interface ProxyRequest {
+  type: "fetchai:proxy-request-v1";
+  id: string;
+  method: Method;
+  args: any[];
+}
+
+export interface ProxyResponse {
+  type: "fetchai:proxy-response-v1";
+  id: string;
+  result: Result | undefined;
+}
+
+function generateNewRequestId(): string {
+  const bytes = new Uint8Array(8);
+  return Array.from(crypto.getRandomValues(bytes))
+    .map((value) => {
+      return value.toString(16);
+    })
+    .join("");
+}
+
+export function createProxyRequest(method: Method, args: any[]): ProxyRequest {
+  return {
+    type: "fetchai:proxy-request-v1",
+    id: generateNewRequestId(),
+    method,
+    args: JSONUint8Array.wrap(args),
+  };
+}
+
+export function createProxyResponse(
+  id: string,
+  result: Result | undefined
+): ProxyResponse {
+  return {
+    type: "fetchai:proxy-response-v1",
+    id,
+    result,
+  };
+}
+
+function isProxyRequest(obj: any): obj is ProxyRequest {
+  return !(!obj || obj.type !== "fetchai:proxy-request-v1" || !obj.id);
```

Comment:
Can this be simplified to:
```suggestion
  return !(!obj?.id || obj.type !== "fetchai:proxy-request-v1");
```

(That's a fun bit of logic to think through there btw :smile:)